### PR TITLE
[export] Move lifted constants out of state dict

### LINKED ISSUE #110484: [export] Constant tensors should not get lifted to buffers
Today, if we have a constant tensor (e.g. generate by a `torch.tensor(0)` call), it will get lifted to be a buffer (https://github.com/pytorch/pytorch/blob/main/torch/_export/passes/lift_constant_tensor_pass.py).

This is potentially problematic—the state dict is a well-established abstraction in PyTorch, and mutating it due to export implementation details may be surprising for users. As a concrete example, we have many users that guard model compatibility based on state dict keys. If export adds a key based on some details of how tracing happened, those compatibility checks will break and users will be confused as to why.

Instead, we should handle tensor constants in the tensor specification. Nodes should be able to take constant tensors as inputs, and we should have a table mapping constant tensor handles to in-archive serialized blobs. The serialization/deserialization systems will need to be updated to properly load/save these tensors.

cc @avikchaudhuri @gmagogsfm @zhxchen17 @tugsbayasgalan @angelayi

### PR DESCRIPTION
**Pull request by angelayi**
Fixes https://github.com/pytorch/pytorch/issues/110484

Instead of storing constants in the state dict we can store them in a separate structure.

### CODE REVIEW

File: test/export/test_export.py [hunk 1]
```diff
@@ -1366,6 +1391,41 @@ def forward(self, x):
...
         self.assertEqual(actual_source_fns, exp_source_fns)
 
+    def test_lifted_constants(self) -> None:
+        def f(x):
+            return x + torch.tensor(3)
+
+        ep = export(f, (torch.tensor(1),))
+
+        for node in ep.graph.nodes:
+            self.assertTrue(node.op != "get_attr")
```
Comment (tugsbayasgalan) [inline test/export/test_export.py:1401]: nit: Could you use self.assertExpectedInline to assert the graph format?

File: test/export/test_export.py [hunk 2]
```diff
@@ -1366,6 +1391,41 @@ def forward(self, x):
...
+        class Foo(torch.nn.Module):
+            def __init__(self):
+                super().__init__()
+                self.a = torch.tensor(3)
+
+            def forward(self, x):
+                list_tensor = [torch.tensor(3), torch.tensor(4)]
+                return x + self.a + list_tensor[0] + list_tensor[1]
+
+        ep = export(Foo(), (torch.tensor(1),))
```
Comment (tugsbayasgalan) [inline test/export/test_export.py:1414]: What happens when you call ep.module? Can you add a test for that too?

File: torch/export/exported_program.py [hunk 3]
```diff
@@ -289,6 +269,17 @@ def buffers(self) -> Collection[str]:
             if isinstance(s.target, str)
         ]
 
+    # A list of lifted constant tensors
+    @property
+    def lifted_tensor_constants(self) -> Collection[str]:
+        # TODO Make this tuple.
```
Comment (zhxchen17) [inline torch/export/exported_program.py:275]: We can make this tuple now?

File: torch/_export/serde/schema.py [hunk 4]
```diff
@@ -304,3 +310,10 @@ class ExportedProgram:
     equality_constraints: List[Tuple[Tuple[str, int], Tuple[str, int]]]
     schema_version: int
     example_inputs: Optional[Tuple[List[bytes], Dict[str, bytes]]]
+
+
+@dataclass
+class SerializedArtifact:
+    serialized_exported_program: bytes
+    serialized_satte_dict: bytes
+    serialized_tensor_constants: bytes
```
Comment (zhxchen17) [inline torch/_export/serde/schema.py:319]: Doesn't seem to belong to schema.py.

File: torch/_export/serde/serialize.py [hunk 5]
```diff
@@ -1585,15 +1571,20 @@ def default(self, obj):
 def serialize(
     exported_program: ep.ExportedProgram,
     opset_version: Optional[Dict[str, int]] = None,
-) -> Tuple[bytes, bytes]:
-    serialized_exported_program, serialized_state_dict = (
+) -> SerializedArtifact:
```
Comment (zhxchen17) [inline torch/_export/serde/serialize.py:1574]: Now sigmoid also uses this interface, you may want to migrate there as well.

File: torch/export/exported_program.py (hunk 3 above)
Comment (angelayi) [reply torch/export/exported_program.py:275]: Idk, I just copied from parameters/buffers above.
